plugin.py

```
@@ -197,6 +230,67 @@ def run_go_command(
     )
 
 
+class GoplsCommand(LspTextCommand):
+    session_name = SESSION_NAME
+
+
+class GoplsRunVulnCheckCommand(GoplsCommand):
+    def run(self, edit: sublime.Edit) -> None:
+        self.edit = edit
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+
+        view = self.view
+        if not view:
+            return
+
+        folders = session.get_workspace_folders()
+        if len(folders) < 1:
+            path = os.path.dirname(uri_from_view(self.view))
+            self.run_gopls_vulncheck(path)
+        elif len(folders) == 1:
+            self.run_gopls_vulncheck(folders[0].uri())
+        else:
+            window = self.view.window()
+            if not window:
+                return
+            window.show_quick_panel(
+                [
+                    sublime.QuickPanelItem(folder.name, folder.uri())
+                    for folder in folders
+                ],
+                on_select=lambda x: self.run_gopls_vulncheck(folders[x].uri())
+                if x != -1
+                else None,
+            )
+
+    def run_gopls_vulncheck(self, path: str) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+
+        session.send_request(
+            Request(
+                'workspace/executeCommand',
+                {'command': 'gopls.run_vulncheck_exp', 'arguments': [{'dir': path}]},
+            ),
+            on_result=lambda x: self.show_results_async(x.get('Vuln')),
+        )
```

Can you add handling for error? I've tried this on some project and it responded with `:: <~~ gopls 4: {'message': 'not implemented', 'code': 0}`.

I'm guessing because my version of `go` is too old but it's not a great experience when it fails silently. Or well, it this message shows in the status field but I would rather expected some more explicit response for user initiated action like that. Possibly a dialog box.

---

Upgraded to go 1.18.5 and it still doesn't work.

```
:: --> gopls workspace/executeCommand(10): {'arguments': [{'dir': 'file:///usr/local/workspace/temp/go-tools/gopls'}], 'command': 'gopls.run_vulncheck_exp'}
:: <-- gopls window/workDoneProgress/create(8): {'token': '6129484611666145821'}
:: >>> gopls 8: None
:: <-  gopls $/progress: {'value': {'title': 'Running vulncheck', 'message': 'Running...', 'cancellable': True, 'kind': 'begin'}, 'token': '6129484611666145821'}
:: <-  gopls $/progress: {'value': {'message': 'failed', 'kind': 'end'}, 'token': '6129484611666145821'}
:: <~~ gopls 10: {'code': 0, 'message': 'not implemented'}
```

What is up with that?

---

Thanks. I'll check this out